### ISSUE TITLE
[OJS 3.0.1] Deleted locale lingers in some settings, causing form failure

### ISSUE THREAD

**Issue by mittinatten** [pkp/pkp-lib#2254]
A journal I administer at some point had the locale `no_NO` active, probably as part of a test. When we upgraded to 3.0 I deleted it from the list of locales, after the upgrade, so that it wouldn't cause confusion, but mainly because there were errors clogging the log. When upgrading from OJS 2 to 3, the locale changes name from `no_NO` to `nb_NO`, which is correct, but apparently not everything is updated, because there where lots of errors of the type

     PHP Notice:  Undefined index: no_NO in ..../lib/pkp/classes/context/Context.inc

Removing the locale however caused at least one form to fail: When I edit an email form it can't be saved, with the following error.

    Errors occurred processing this form
    A subject is required for the email.
    A body is required for the email.

When I click on a input field it expands to two fields (I assume for two locales), and if I fill in both both subject and message I can save the template.

I grepped a database dump for `no_NO` and found it in a bunch of places (in some of those it probably makes sense not to delete the data for the locale in case it is reactivated). The offending setting was in `journal_settings` where both `settingsupportedSubmissionLocales` and `supportedFormLocales` had the value `a:2:{i:0;s:5:"en_US";i:1;s:5:"no_NO";}`. If I change this to `a:1:{i:0;s:5:"en_US";}` the error disappears. 

**In conclusion**

1. When you deactive/delete a locale all settings aren't updated properly. 
2. The migration from `no_NO` to `nb_NO` in the upgrade script doesn't cover all relevant tables.

**Comment by NateWr** [comment]
I believe this is a duplicate of https://github.com/pkp/pkp-lib/issues/6938.
